Quaternion.mpl

```diff
@@ -25,6 +30,44 @@ quaternion: [{
   entries: new;
 }];
 
+quaternion: [
+  m:;
+  m matrix?
+  m getColCount 3 = and
+  m getRowCount 3 = and
+] [
+  m: trans;
+  trace: 3 [i i m @ @] times + +;
+  trace 0 trace cast > [
+    a: trace 1 trace cast + sqrt;
+    b: 0.5 a cast a /;
+    (
+      1 2 m @ @ 2 1 m @ @ - b *
+      2 0 m @ @ 0 2 m @ @ - b *
+      0 1 m @ @ 1 0 m @ @ - b *
+      0.5 a cast a *
+    ) quaternion
+  ] [
+    next: (1 2 0);
+    i: 1 1 m @ @ 0 0 m @ @ > [1] [0] if ;
+    2 2 m @ @ i i m @ @ > [
+     2 !i
+    ] when
+    j: i next @;
+    k: j next @;
+
+    q: trace Quaternion;
+    a: i i m @ @ j j m @ @ - k k m @ @ - 1 trace cast + sqrt;
+    b: 0.5 a cast a /;
+
+    0.5 a cast a *            i @q @ set
+    i j m @ @ j i m @ @ + b * j @q @ set
+    i k m @ @ k i m @ @ + b * k @q @ set
+    j k m @ @ k j m @ @ - b * 3 @q @ set
```

Comment:
I tested it on 100 random unit quaternions `q` this way  `q2: q matrix quaternion`; this condition `L1(q2-q) < 1e-6` is valid, do you want something like this in tests/QuaternionTest.mpl or just some handwritten quaternions? You can also check [this link](https://dev.gravilink.com/Gamedev/snake_cpp/src/branch/master/snake/client/universal/base/algebra.h#L45), it was taken from there

Reply:
Yes, i wanted something like this. I verified that this code is equivalent to the one from a working project, so everything must be ok. The test in tests/QuaternnionTest.mpl is nice, but not required anymore. I only want `a` and `b` to be renamed accordingly to the base code. Fix it and the other comments and i will approve the PR.

Reply:
There is only `fr` variable in the base code, instead of `a` and `b`, i did so because of quaternion creation looks better, and for me `fr` is meaningless coefficient (from `fraction`), but maybe i am wrong, not like `trace`, for example, so `a` and `b` are not worse. I can rename `b` to `fr`, but what with `a` then? What do you think?

Reply:
I think the best is to reference `w` component of a quaternion, the real part.